app/controllers/charges_controller.rb

```diff
@@ -3,8 +3,8 @@ def new
   end
 
   def create
-    @total_amount = 50
-    @amount = (@total_amount*100)
+    @total_amount = Order.last.total
+    @amount = @total_amount.to_i*100
```

Comment:
Do you really need the `to_i`?  I was under the impression that `total` returns an integer...